src/tv2-common/helpers/serverResume.ts

```diff
@@ -0,0 +1,173 @@
+import {
+	IActionExecutionContext,
+	IBlueprintActionTriggerMode,
+	IBlueprintPartInstance,
+	IBlueprintResolvedPieceInstance,
+	VTContent
+} from '@tv2media/blueprints-integration'
+import { PartEndStateExt, t } from 'tv2-common'
+import { SharedSourceLayers } from 'tv2-constants'
+import _ = require('underscore')
+import { DVEPieceMetaData } from '../content'
+
+/** If the seek position is closer to the end of the file than this value, it will be reset to 0. */
+const REMAINING_DURATION_BEFORE_RESET = 2 * 1000
+
+export interface ServerPosition {
+	fileName: string
+	lastEnd: number
+	isPlaying: boolean
+	endedWithPartInstance?: string
+}
+
+export enum ServerSelectMode {
+	RESUME = 'resume',
+	RESET = 'reset'
+}
+
+export function getServerSeek(
+	lastServerPosition: ServerPosition | undefined,
+	fileName: string,
+	mediaObjectDuration?: number,
+	triggerMode?: string
+): number {
+	if (triggerMode === ServerSelectMode.RESET) {
+		return 0
+	}
+	if (lastServerPosition?.fileName === fileName && !lastServerPosition.isPlaying) {
+		if (!mediaObjectDuration) {
+			return lastServerPosition.lastEnd
+		}
+		const seek = lastServerPosition.lastEnd % mediaObjectDuration
+		const remainingDuration = mediaObjectDuration - seek
+		if (remainingDuration > REMAINING_DURATION_BEFORE_RESET) {
+			return seek
+		}
+	}
+	return 0
+}
+
+export async function getServerPosition(
+	context: IActionExecutionContext,
+	replacingCurrentPieceWithOffset?: number
+): Promise<ServerPosition | undefined> {
+	const partInstance = await context.getPartInstance('current')
+	if (!partInstance) {
+		throw new Error('Missing current PartInstance while calculating serverOffsets')
+	}
+
+	const pieceEnd =
+		replacingCurrentPieceWithOffset !== undefined
+			? context.getCurrentTime() + replacingCurrentPieceWithOffset
+			: undefined
+
+	return getServerPositionForPartInstance(partInstance, await context.getResolvedPieceInstances('current'), pieceEnd)
+}
+
+/**
+ * Calculate the offsets for clips based on the pieceinstances that have already been played.
+ */
+export function getServerPositionForPartInstance(
+	partInstance: IBlueprintPartInstance,
+	pieceInstances: IBlueprintResolvedPieceInstance[],
+	setCurrentPieceToNow0?: number
```

Comment:
Why include the 0 in the name?
Also, prefixing something with 'set' indicates that it is a method that does something. Here the parameter is a number, so using the parameter doesn't 'do' something. (It is the same princible with prefixing something with 'is' or 'has'. You useally would only do that for booleans or methods that returns a boolean)